src/data/battler-tags.ts

```diff
@@ -1453,6 +1453,22 @@ export class FireSpinTag extends VortexTrapTag {
   }
 }
 
+/**
+ * Used for G-Max Centiferno that leaves a fire spin
+ * that persists even on the user leaving the field
+ */
+export class GMaxFireSpinTag extends FireSpinTag {
+  constructor(turnCount: number, sourceId: number) {
+    super(turnCount, sourceId);
+    this.tagType = BattlerTagType.G_MAX_FIRE_SPIN;
+    this.sourceMove = Moves.G_MAX_CENTIFERNO;
+  }
+
+  override isSourceLinked(): boolean {
+    return false;
+  }
```

Comment:
```suggestion
  }

  override canAdd(pokemon: Pokemon): boolean {
    return !pokemon.getTag(TrappedTag);
  }
```
It turns out that `DamagingTrapTag` has its own Substitute check that needs to be removed for the G-Max tags.